packages/grafana-ui/src/components/CustomScrollbar/CustomScrollbar.tsx

```diff
@@ -56,15 +46,24 @@ export const CustomScrollbar: FC<Props> = ({
     }
   };
 
-  const updateAfterMount = () => {
-    if (!ref || !ref.current) {
-      return;
-    }
-    const scrollbar = ref.current as any;
-    if (scrollbar.update) {
-      scrollbar.update();
-    }
-  };
+  useEffect(() => {
+    updateScroll();
+  });
+
+  /**
+   * Special logic for doing a update a few milliseconds after mount to check for
+   * updated height due to dynamic content
+   */
+  if (updateAfterMountMs) {
+    useEffect(() => {
+      setTimeout(() => {
+        const scrollbar = ref.current as any;
+        if (scrollbar?.update) {
+          scrollbar.update();
+        }
+      }, updateAfterMountMs);
+    }, []);
+  }
```

Comment:
Doesn't this break the rules of hooks? [https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)

Reply:
How do I do an onMount (only on mount ) one time effect otherwise?

Reply:
Which rule does it break?

Reply:
`Don’t call Hooks inside loops, conditions, or nested functions.`

Reply:
This looks like it's calling a hook inside a condition.

Reply:
ah, true. Wonder if it still is ok. Just did not want this onMount effect if was not needed

Reply:
I think this should be fine, as updateAfterMountMs should never change during a scrollbars lifetime